### ISSUE TITLE
False positive with Lint/UselessRescue

### ISSUE THREAD

**Issue by jesse-shopify** [rubocop/rubocop#11420]
## Expected behavior

- Should not trigger cop when exception variable is not used in catch section but is used in ensure section

## Actual behavior

- Cop is triggered

## Steps to reproduce the problem

```ruby
def false_positive
  puts "Method which can raise goes here"
rescue => e
  raise
ensure
  puts e
end
```

## RuboCop version

```
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.1) [arm64-darwin21]
  - rubocop-graphql 0.18.0
  - rubocop-performance 1.15.2
  - rubocop-rails 2.17.4
  - rubocop-sorbet 0.6.11
```
